File: frontend/packages/lib-shared/api/modules/adUpstreamAgent.ts
import type { CordysAxios } from '@lib/shared/api/http/Axios';
import { AdUpstreamAgentUrl, AdUpstreamAgentPageUrl } from '@lib/shared/api/requrls/adUpstreamAgent';
import type { CommonList } from '@lib/shared/models/common';

export interface AdUpstreamAgentItem {
    id: string;
    name?: string;
    creditCode?: string;
    signingEntity?: string;
    contactPerson?: string;
    contactPhone?: string;
    cooperationStatus?: number;
    status?: number;
    statusLabel?: string;
    businessEntityId?: string;
    businessEntityName?: string;
    remark?: string;
    createTime?: number;
}

export interface AdUpstreamAgentSaveParams {
    id?: string;
    name?: string;
    creditCode?: string;
    signingEntity?: string;
    contactPerson?: string;
    contactPhone?: string;
    cooperationStatus?: number;
    status?: number;
    remark?: string;
    businessEntityId?: string;
}

type AdUpstreamAgentPageParams = Record<string, any>;
type AdUpstreamAgentPageResult = CommonList<AdUpstreamAgentItem>;

export default function useAdUpstreamAgentApi(CDR: CordysAxios) {
    function createAdUpstreamAgent(data: AdUpstreamAgentSaveParams) {
        return CDR.post<any>({ url: AdUpstreamAgentUrl, data });
    }
    function updateAdUpstreamAgent(data: AdUpstreamAgentSaveParams) {
        return CDR.put<any>({ url: AdUpstreamAgentUrl, data });
    }
    function getAdUpstreamAgentDetail(id: string) {
        return CDR.get<any>({ url: `${AdUpstreamAgentUrl}/${id}` });
    }
    function getAdUpstreamAgentPage(data: AdUpstreamAgentPageParams) {
        return CDR.post<AdUpstreamAgentPageResult>({ url: AdUpstreamAgentPageUrl, data }, { ignoreCancelToken: true });
    }
    function deleteAdUpstreamAgent(id: string) {
        return CDR.delete({ url: `${AdUpstreamAgentUrl}/${id}` });
    }
    return { createAdUpstreamAgent, updateAdUpstreamAgent, getAdUpstreamAgentDetail, getAdUpstreamAgentPage, deleteAdUpstreamAgent };
}

File: backend/crm/src/main/java/cn/cordys/crm/ad/upstreamagent/domain/AdUpstreamAgent.java
package cn.cordys.crm.ad.upstreamagent.domain;

import cn.cordys.common.domain.BaseModel;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Table(name = "ad_upstream_agent")
public class AdUpstreamAgent extends BaseModel {

    @Schema(description = "代理名称")
    private String name;

    @Schema(description = "社会信用代码")
    private String creditCode;

    @Schema(description = "签约主体")
    private String signingEntity;

    @Schema(description = "联系人")
    private String contactPerson;

    @Schema(description = "电话")
    private String contactPhone;

    @Schema(description = "合作状态:10正常/20停用")
    private Integer cooperationStatus;

    @Schema(description = "状态:10正常/20停用")
    private Integer status;

    @Schema(description = "备注")
    private String remark;

    @Schema(description = "业务主体")
    private String businessEntityId;

    @Schema(description = "组织(租户)id")
    private String organizationId;

    @Schema(description = "是否删除:0-否/1-是")
    private Integer deleted = 0;
}
